src/data-table/types.ts

```diff
@@ -0,0 +1,73 @@
+import { CSSProperties, ReactNode } from "react";
+
+export interface DataTableProps {
+    testid?: string | undefined;
+    headersConfig?: HeadersConfigProps | undefined;
+    headers: HeadersProps[];
+    rowsConfig?: RowsConfigProps | undefined;
+    rows?: RowProps[] | undefined;
+    className?: string | undefined; // css class to put on table
+    selectionConfig?: SelectionConfigProps | undefined;
+    selection?: string[] | undefined; // ids of all selected rows
+    sortIndicators?: SortIndicatorsProps | undefined; // columns that want to show a sort indicator
+    actionsConfig?: ActionsConfigProps | undefined;
+    alternatingRows?: boolean | undefined;
+    customEmptyView?: () => ReactNode | string | undefined;
+    isLoadingData: boolean | undefined;
```

Comment:
some flexibility for other potential statuses like error in the future
```suggestion
    loadState?: "loading" | "success" | undefined;
```